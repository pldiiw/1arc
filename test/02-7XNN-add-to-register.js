'use strict';

const engine = require('engine').initialize();
const addToRegister = require('instructions').addToRegister;

const engine1 = addToRegister(engine, 1, 20);
const engine2 = addToRegister(engine1, 1, 80);

const engine3 = addToRegister(engine, 3, 300);
const engine4 = addToRegister(engine3, 3, 260);
const engine5 = addToRegister(engine3, 3, 10);


console.log(engine1.data[1]);
console.log(engine2.data[1]);
console.log(engine2.data[0xF]);

console.log(engine3.data[3]);
console.log(engine3.data[0xF]);
console.log(engine4.data[3]);
console.log(engine4.data[0xF]);
console.log(engine5.data[3]);
console.log(engine5.data[0xF]);
