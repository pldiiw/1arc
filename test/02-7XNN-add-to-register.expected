20
100
0
44
1
48
1
58
0
